selfdrive/car/hyundai/interface.py

```diff
@@ -0,0 +1,266 @@
+from selfdrive.car.hyundai.camstate import CamState, get_can_parser2
+from selfdrive.car.hyundai.values import CAR
+from selfdrive.swaglog import cloudlog
+
+try:
+  from selfdrive.car.hyundai.carcontroller import CarController
+except ImportError:
+  CarController = None
+
+
+class CarInterface(object):
+  def __init__(self, CP, sendcan=None):
+    self.CP = CP
+    self.VM = VehicleModel(CP)
+    self.idx = 0
+    self.lanes = 0
+    self.lkas_request = 0
+    
+    self.frame = 0
+    self.gas_pressed_prev = False
+    self.brake_pressed_prev = False
+    self.can_invalid_count = 0
+    self.cruise_enabled_prev = False
+
+    # *** init the major players ***
+    self.CS = CarState(CP)
+    self.CamS = CamState(CP)
+
+    self.cp = get_can_parser(CP)
+    self.cp2 = get_can_parser2(CP)
+
+    # sending if read only is False
+    if sendcan is not None:
+      self.sendcan = sendcan
+      self.CC = CarController(self.cp.dbc_name, CP.carFingerprint, CP.enableCamera)
+
+  @staticmethod
+  def compute_gb(accel, speed):
+    return float(accel) / 3.0
+
+  @staticmethod
+  def calc_accel_override(a_ego, a_target, v_ego, v_target):
+    return 1.0
+
+  @staticmethod
+  def get_params(candidate, fingerprint):
+
+    # kg of standard extra cargo to count for drive, gas, etc...
+    std_cargo = 200 # Fuel = 70kg + Driver = 90kg + Roofracks, towball, etc = 40kg
+
+    ret = car.CarParams.new_message()
+
+    ret.carName = "hyundai"
+    ret.carFingerprint = candidate
+
+    ret.safetyModel = car.CarParams.SafetyModels.hyundai
+
+    # pedal
+    ret.enableCruise = False
+
+    rotationalInertia = 2500
+
+    tireStiffnessFront = 85400
+    tireStiffnessRear = 90000
+
+    ret.steerKiBP, ret.steerKpBP = [[0.], [0.]]
+    ret.steerActuatorDelay = 0.1  # Default delay, Prius has larger delay
+
+    #borrowing a lot from corolla, given similar car size
+    ret.steerKf = 0.00008   # full torque for 20 deg at 80mph means 0.00007818594
+    ret.steerRateCost = 1.
+    stop_and_go = True
+    ret.mass = 1985 + std_cargo
+    ret.wheelbase = 2.78 
+    ret.steerRatio = 15.0
+    ret.steerKpV, ret.steerKiV = [[0.20], [0.007]]
+    ret.centerToFront = ret.wheelbase * 0.4
+
+    ret.longPidDeadzoneBP = [0., 9.]
+    ret.longPidDeadzoneV = [0., .15]
+
+    # min speed to enable ACC. if car can do stop and go, then set enabling speed
+    # to a negative value, so it won't matter.
+    ret.minEnableSpeed = -1.
+
+    centerToRear = ret.wheelbase - ret.centerToFront
+    # TODO: get actual value, for now starting with reasonable value for
+    # civic and scaling by mass and wheelbase
+    ret.rotationalInertia = rotationalInertia * ret.mass * ret.wheelbase**2
+
+    # TODO: start from empirically derived lateral slip stiffness for the civic and scale by
+    # mass and CG position, so all cars will have approximately similar dyn behaviors
+    ret.tireStiffnessFront = tireStiffnessFront * ret.mass * (centerToRear / ret.wheelbase)
```

Comment:
@emmertex this is an important bug: seem my port for correct dimensional scaling.
As is, car should strongly underestimate the needed steering angle to make a turn